### DISCUSSION TITLE
Package state iOS 13 support

### DISCUSSION THREAD

**Discussion by gsl-anthonymerle** [pointfreeco/swift-clocks#14] [General]
Hello 👋 

So the Package.swift file states that the swift-clock framework is supported for iOS 13
I don't get why is that, as the Clock type is only available from iOS 16

For now we're only able to add the package on a project (mine's minimum version target is iOS 14), but we can't use any piece of it unless we specify and iOS 16 availability

If it's only a mistake, could you update the Package.swift file accordingly ?

**Comment by stephencelis** [comment]
We do this because we want libraries to be able to conditionally bring in functionality using `if #available` checks without having to change the minimum deployment target of the depending library. For example, [the Composable Architecture](/pointfreeco/swift-composable-architecture) depends on this library to support controlling clocks in iOS 16+ apps, but the library itself more generally supports iOS 13.

If there's a better way to do this, though, and you can point us to an example in the wild, please let us know!

**Reply by gsl-anthonymerle** [reply]
Ah yes ! It makes total sense, thank you for your (quick) answer 🙏 
Unfortunately I have no idea how to achieve this capability, but because I understand it now it doesn't look like an issue to me anymore
